### ISSUE TITLE
False positive with unneeded_break_in_switch while in a loop

### ISSUE THREAD

**Issue by nicolasgoutaland** [realm/SwiftLint#3724]
### New Issue Checklist

- [X ] Updated SwiftLint to the latest version
- [ X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The **unneeded_break_in_switch** is triggering false positive in some cases.

##### Complete output when running SwiftLint, including the stack trace and command used

```
AppMigration.swift:133:25: warning: Unneeded Break in Switch Violation: Avoid using unneeded break statements. (unneeded_break_in_switch)```

### Environment

* SwiftLint version (0.44.0)
* Installation method used (CocoaPods)
* Paste your configuration file:

```yml
Using default configuration
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using (check `xcodebuild -version`)? 12.5.1, but same issue with latest Xcode version (13)
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
            let items = [Int]()
            for item in items {
                if bar() {
                    do {
                        try foo()
                    } catch {
                        bar()
                        break
                    }
                }
            }
```

```swift
// This doesn't triggers a violation:
            let items = [Int]()
            for item in items {
                if bar() {
                    do {
                        try foo()
                    } catch {
//                        bar()
                        break
                    }
                }
            }
```
